src/__experimental__/components/numeral-date/numeral-date.component.js

```diff
@@ -0,0 +1,165 @@
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import invariant from 'invariant';
+import Events from '../../../utils/helpers/events';
+import DateHelper from '../../../utils/helpers/date';
+import StyledNumeralDate, { StyledDateField } from './numeral-date.style';
+import Textbox from '../textbox';
+
+const NumeralDate = ({
+  dateFormat,
+  defaultValue,
+  errorMessage,
+  // errorState,
+  id,
+  name,
+  onBlur,
+  onChange,
+  placeholder,
+  value
+}) => {
+  const isControlled = value !== undefined;
+  const initialValue = isControlled ? value : defaultValue;
+  invariant(
+    typeof initialValue === 'string',
+    'This component has no initial value'
+  );
+
+  let inputRef = useRef();
+  const [isActive, setIsActive] = useState(inputRef.current === document.activeElement);
+  const [errorState, setErrorState] = useState(false);
+  const [dateValue, setDateValue] = useState({
+    dd: '', mm: '', yyyy: ''
+  }); // value date textbox field
+
+  // For validating dates
+  const handleDateValidation = () => {
+    const { dd, mm, yyyy } = dateValue;
+    const dateIsEmpty = !yyyy.length || !mm.length || !dd.length;
+    const dateValueString = dateIsEmpty ? '' : `${yyyy}-${mm}-${dd}`;
+
+    if (!dateIsEmpty) {
+      if (!DateHelper.isValidDate(dateValueString)) {
+        setErrorState(true);
+      } else setErrorState(false);
+    }
+  };
+
+  const isValidKeyPress = (ev) => {
+    if (Events.isNumberKey(ev) || Events.isDeletingKey(ev) || Events.isTabKey(ev)) {
+      return true;
+    }
+    return false;
+  };
+
+  const onKeyPress = (ev) => {
+    if (!isValidKeyPress(ev)) {
+      ev.preventDefault();
+    }
+  };
+
+  const handleChange = (e, itemId) => {
+    if (onChange) {
+      onChange(e);
+    }
+    if (e.target.value !== dateValue[itemId]) {
+      setDateValue({ ...dateValue, [itemId]: e.target.value });
+    }
+  };
+
+  const handleOnFocus = () => {
+    setIsActive(true);
+  };
+
+  const handleBlur = (ev) => {
+    setIsActive(false);
+    handleDateValidation();
+    if (onBlur) {
+      onBlur({ name: ev.target.name, id: ev.target.id, value: { ...dateValue } });
+    }
+  };
+
+
+  return (
+    <>
+      <StyledNumeralDate
+        name={ name }
+        id={ id }
+        isActive={ isActive }
+        onBlur={ handleBlur }
+        onFocus={ handleOnFocus }
+        onKeyPress={ onKeyPress }
+      >
+        {
+          dateFormat.map((datePart, i) => (
```

Comment:
could you change the `i` argument to be less vague?

Reply:
@grabkowski - is this comment still valid from the original WIP review you did 10 days ago?

Reply:
yes